tests/geometry/WorkingGeometry.spec.ts

```diff
@@ -324,19 +340,24 @@ describe('WorkingGeometry', () => {
 
             expect(square.vertices).toEqualArrVec4(TestHelper.square().vertices);
         });
-        it('will throw an error when it is skewed', () => {
+        it('will perform a rotation when the pull and destination points are not colinear', () => {
```

Comment:
I'm seeing a rotation and translation as well? The square translated to (1, 1, 0) and then rotated -90 degrees on the z-axis. I feel like this isn't expected behaviour, I was expecting the scale to flatten the square to a line with 4 points with 2 each on (1, 0, 0) and (0, 0, 0).

Can you help me visualize this?

Reply:
(I'm omitting the z coordinate for clarity)
So this is holding the bottom-left corner of the square and pulling the bottom-right corner to (1,1), while allowing the square to be skewed in the process. However, after translating the bottom-left corner to be the origin, the pull point becomes (-1, 0) and the destination point becomes (0, 1), and the entire function resolves to becoming a -90 degree rotation about (1,0). Does that make sense?

Though this is a good point to bring up, since calling this function `scale` doesn't make much sense. It might be better to rename the pair of scaling functions (based on the previous two threads) as follows:

- `scale` -> `stretchTo`
- `scaleByFactor` -> `rigidStretchTo`

What do you think?

Reply:
Err, maybe s/`rigidStretchTo`/`rigidPullTo`? That could make it a bit clearer that there's no non-uniform scaling going on in that function.

Reply:
I think the distinction between "freeform" vs "proportional" stretching is good! How about `freeformStretchTo` and `proportionalStretchTo`? That way the names are consistently structured, a user can see the similarities and differences between the two functions.

Reply:
That sounds good! What do you think about possibly changing the ordering of the words? I'm just thinking that it's easier to search up `stretchFreeformTo` and `stretchProportionalTo` since they start with the same letters. But that's a nit which we can easily change later, and I'm also not a fan of the `stretchPropotionalTo` since I think the meaning is distorted. 🤔

Reply:
I think `.*StretchTo` should still be easy to search for, right? If it isn't, we could file an umbrella issue for online docs and add this search grouping as a feature we want.

Reply:
Personal opinion: if we ever implement some form of auto completion in an editor putting all stretching functions with the same prefix would be useful. Also since our target audience is people who are not familiar with programming environments, I would prefer to not have them to reach for regexs :)

Reply:
@abhimadan did you mean that the user will type in `.*StretchTo`? I'm unfamiliar with regexes myself but I would imagine the user to start typing `stretch` and then expect to see the functions they're looking for. As long as they're using an editor that does fuzzy search typeahead suggestions then it shouldn't be a problem :)

Reply:
(So `.` is a wildcard character in most regex engines, and `*` matches the preceeding character zero or more times, so `.*` basically means any string.)

My bad, I was thinking of users searching through an online docs page, not about editor autocompletion.  In the former case, I don't think implicitly doing a `.*[searchtext].*` on every search would be a problem, and I'm not familiar enough with how editor autocomplete plugins work to comment on the latter. I don't want to commit too much to names right now since it should be simple enough to change them later, but I'm not a huge fan of the `stretch.*To` names as alternatives since they seem to distort the meaning of the names a bit.